osmdroid-shape/src/main/java/org/osmdroid/shape/DefaultShapeMetaSetter.java

```diff
@@ -0,0 +1,74 @@
+package org.osmdroid.shape;
+
+import android.graphics.Paint;
+
+import net.iryndin.jdbf.core.DbfRecord;
+
+import org.osmdroid.views.overlay.Marker;
+import org.osmdroid.views.overlay.Overlay;
+import org.osmdroid.views.overlay.PolyOverlayWithIW;
+import org.osmdroid.views.overlay.Polygon;
+
+import java.nio.charset.Charset;
+import java.text.ParseException;
+
+public class DefaultShapeMetaSetter implements ShapeMetaSetter {
+
+    private String metaString;
+    private String subString;
+    private Overlay mOverlay;
+    private String shapeType;
+
+    @Override
+    public void set(DbfRecord metadata, Overlay overlay) throws ParseException {
+        if (metadata != null){
+            metadata.setStringCharset(Charset.defaultCharset());
+            metaString = metadata.toMap().toString();
+        } else{
+            metaString = "";
+        }
+
+        shapeType = overlay.getClass().getName();
```

Comment:
`shapeType = overlay.getClass().getName()` is very ugly; this is not OOP.
Just implement
* `public void set(DbfRecord metadata, Polygon polygon) throws ParseException`
* `public void set(DbfRecord metadata, Polyline polyline) throws ParseException`
* `public void set(DbfRecord metadata, Marker marker) throws ParseException`
In those methods you compute whatever is relevant for each of the 3 possible Overlay classes, and you set it directly.